### PR TITLE
Switch publish script into smoke tests in rules yaml

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kubernetes/publishing-bot/issues/26

### CODE REVIEW

File: cmd/publishing-bot/main.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func main() {
 	repoName := flag.String("source-repo", "", "the name of the source repository (eg. kubernetes)")
 	repoOrg := flag.String("source-org", "", "the name of the source repository organization, (eg. kubernetes)")
 	targetOrg := flag.String("target-org", "", `the target organization to publish into (e.g. "k8s-publishing-bot")`)
-	basePublishScriptPath := flag.String("base-publish-script-path", "./publish_scripts", `the base path in source repo where bot will look for publishing scripts`)
+	basePublishScriptPath := flag.String("base-publish-script-path", "/publish_scripts", `the base path in source repo where bot will look for publishing scripts`)
```
Comment (mfojtik) [inline cmd/publishing-bot/main.go]: i think this was meant to be a relative path inside the repo

File: cmd/publishing-bot/config/rules.go [hunk 2]
```diff
@@ -48,8 +48,8 @@ type BranchRule struct {
 type RepositoryRule struct {
 	DestinationRepository string       `yaml:"destination"`
 	Branches              []BranchRule `yaml:"branches"`
-	// if empty (e.g., for client-go), publisher will use its default publish script
-	PublishScript string `yaml:"publish-script"`
+	SmokeTest             string       `yaml:"smoke-test"` // a multiline bash script
+	Library               bool         `yaml:"library"`
```
Comment (mfojtik) [inline cmd/publishing-bot/config/rules.go]: omitempty?

**Comment by mfojtik** [comment]
one nit, otherwise LGTM

File: cmd/publishing-bot/main.go (hunk 1 above)
Comment (sttts) [reply cmd/publishing-bot/main.go]: It's relative to the current working dir. I think before this PR it was coincidently the root of the container. Will check. I think we have to restore the current working dir correctly. Then the `./` is fine.

File: cmd/publishing-bot/config/rules.go (hunk 2 above)
Comment (sttts) [reply cmd/publishing-bot/config/rules.go]: well, default is false. But printing it (if we ever print rules at all ;-) ) makes sense.

Comment (mfojtik) [reply cmd/publishing-bot/config/rules.go]: generating (serializing) is ugly when we default empty fields ;-)

File: cmd/publishing-bot/main.go (hunk 1 above)
Comment (sttts) [reply cmd/publishing-bot/main.go]: fixed

File: cmd/publishing-bot/config/rules.go (hunk 2 above)
Comment (sttts) [reply cmd/publishing-bot/config/rules.go]: fixed

**Pull request merged**